bitcoin/src/blockdata/transaction.rs

```diff
@@ -686,19 +722,58 @@ impl Transaction {
     /// and can therefore avoid this ambiguity.
     #[inline]
     pub fn weight(&self) -> Weight {
-        let inputs = self.input.iter().map(|txin| {
-            InputWeightPrediction::new(
-                txin.script_sig.len(),
-                txin.witness.iter().map(|elem| elem.len()),
-            )
-        });
-        let outputs = self.output.iter().map(|txout| txout.script_pubkey.len());
-        predict_weight(inputs, outputs)
+        // This is the exact definition of a weight unit, as defined by BIP-141 (quote above).
+        let wu = self.base_size() * 3 + self.total_size();
```

Comment:
Is it possible that this could overflow?

Reply:
Wow sloppy work by me when I re-did this version of the PR, all math operations  in the PR should be handled the same way. Will fix, thanks.